src/App.tsx

```diff
@@ -1,60 +1,95 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { getUsers } from './utils/fetchClient';
+import { User } from './types/User';
+import { PostsList } from './components/PostsList';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [showSideBar, setShowSideBar] = useState(false);
+  const [users, setUsers] = useState<User[] | null>();
+  const [dropDownOpen, setDropDownOpen] = useState(false);
+  const [userSelectedId, setUserSelectedId] = useState(0);
+  const [postSelectedId, setPostSelectedId] = useState(0);
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [isShowForm, setIsShowForm] = useState(false);
+
+  useEffect(() => {
+    getUsers()
+      .then(usersFromServer => {
+        setUsers(usersFromServer);
+      });
+  }, []);
+
+  useEffect(() => {
+    setIsShowForm(false);
+  }, [postSelectedId, userSelectedId]);
```

Comment:
Are you sure that you need to set it to false every time when these values (postSelectedId, userSelectedId) are changed? Maybe add some checks as well?

Reply:
I'm sure